tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -55,6 +55,232 @@ def _emit_event(self, resource, curr_value, msg, instance):
+                    subvol_count += 1
+                except etcd.EtcdKeyNotFound:
+                    break
+            vol_usable_capacity = 0
+            vol_used_capacity = 0
+            vol_pcnt_used = 0
+            if up_bricks == len(bricks):
+                if int(volume.disperse_count) == 0:
+                    # This is distribute or replicate volume
+                    vol_usable_capacity = raw_capacity / int(volume.replica_count)
+                    vol_used_capacity = raw_used / int(volume.replica_count)
+                else:
+                    # this is a disperse volume, with all bricks online
+                    # assumption : all bricks are the same size
+                    disperse_yield = \
+                        float(
+                            int(volume.disperse_count) - int(volume.redundancy_count)
+                        ) / int(volume.disperse_count)
+                    vol_usable_capacity = raw_capacity * disperse_yield
+                    vol_used_capacity = raw_used * disperse_yield
+            else:
+                if int(volume.replica_count) > 1 or int(volume.disperse_count) > 0:
+                    for subvol in subvols:
+                        for brick in subvol:
+                            if brick.status == "Started":
+                                vol_usable_capacity += brick.utilization['total']
+                                vol_used_capacity += brick.utilization['used']
+                            # For replicate volume use only one replica
+                            if int(volume.replica_count) > 1:
+                                break
+                else:
+                    vol_usable_capacity = raw_capacity
+                    vol_used_capacity = raw_used
+            if vol_usable_capacity > 0:
+                vol_pcnt_used = (vol_used_capacity / float(vol_usable_capacity)) * 100
+            volume.usable_capacity = vol_usable_capacity
+            volume.used_capacity = vol_used_capacity
+            volume.pcnt_used = str(vol_pcnt_used)
+            volume.save()
+            cluster_usable_capacity += volume.usable_capacity
+        cluster_pcnt_used = 0
+        if cluster_usable_capacity > 0:
+            cluster_pcnt_used = (cluster_used_capacity / float(cluster_usable_capacity)) * 100
+
+        NS.gluster.objects.Utilization(
+            raw_capacity=cluster_raw_capacity,
+            used_capacity=cluster_used_capacity,
+            usable_capacity=cluster_usable_capacity,
+            pcnt_used=str(cluster_pcnt_used)
+        ).save()
+
+    def _sync_cluster_status(self, volumes):
+        status = 'healthy'
+
+        # Calculate status based on volumes status
+        if len(volumes) > 0:
+            volume_states = self._derive_volume_states(volumes)
+            for vol_id, state in volume_states.iteritems():
+                if 'down' in state or 'partial' in state:
+                    status = 'unhealthy'
```

Comment:
It might be healthy due to volume states, but still we need to look for peers connected state. Thats why it continues below. It same logic from `gstatus`